### PR TITLE
Create Stoa's Transactions Pool DB Schema Part1

### PR DESCRIPTION
**Pull request by TrustHenry**
This transaction pool stores transactions received from Agora.
It will be erased from the pool when it is later saved in the block.

### CODE REVIEW

File: docs/Database_Schema.md [hunk 1]
```diff
@@ -224,3 +224,74 @@ CREATE TABLE IF NOT EXISTS information (
...
+----
+
+## 5. Table **txOutputPool**
+
+### _Schema_
+
+| Column            | Data Type | PK | Not NULL | Default  |Description|
+|:----------------- |:--------- |:--:|:--------:| -------- | --------- |
+|  tx_hash          | BLOB      | Y  | Y        |          | The hash of transaction |
+|  output_index     | INTEGER   | Y  | Y        |          | The index of output in the outputs.|
```
Comment (MichaelKim20) [inline docs/Database_Schema.md]: Why do we need output_index?

File: docs/Database_Schema.md [hunk 2]
```diff
@@ -224,3 +224,74 @@ CREATE TABLE IF NOT EXISTS information (
...
+----
+
+## 4. Table **txInputPool**
+
+### _Schema_
+
+| Column            | Data Type | PK | Not NULL | Default  |Description|
+|:----------------- |:--------- |:--:|:--------:| -------- | --------- |
+| tx_hash           | BLOB      | Y  | Y        |          | The hash of transaction|
+| utxo              | BLOB      | Y  | Y        |          | The hash of the UTXO to be spent|
```
Comment (MichaelKim20) [inline docs/Database_Schema.md]: Why does this(utxo) have to be the key?

File: docs/Database_Schema.md [hunk 3]
```diff
@@ -224,3 +224,74 @@ CREATE TABLE IF NOT EXISTS information (
...
+
+----
+
+## 9. Table **transactionPool**
+
+### _Schema_
+
+| Column            | Data Type | PK | Not NULL | Default  |Description|
+|:----------------- |:--------- |:--:|:--------:| -------- | --------- |
+| tx_hash           | BLOB      | Y  | Y        |          | The hash of transaction |
```
Comment (MichaelKim20) [inline docs/Database_Schema.md:236]:
```suggestion
| time              | INTEGER   |    | Y        |          | Receive TimeStamp |
| tx_hash           | BLOB      | Y  | Y        |          | The hash of transaction |
```
Don't we need the time received?

**Review by MichaelKim20 — COMMENTED** [review]
I have some questions.

File: docs/Database_Schema.md (hunk 3 above)
Comment (TrustHenry) [reply docs/Database_Schema.md:236]: Yes, we need it.

File: docs/Database_Schema.md (hunk 1 above)
Comment (TrustHenry) [reply docs/Database_Schema.md]: The outputs is an array.
A key is required to distinguish.

File: docs/Database_Schema.md (hunk 2 above)
Comment (TrustHenry) [reply docs/Database_Schema.md]: Isn't it unique?

File: docs/Database_Schema.md (hunk 1 above)
Comment (MichaelKim20) [reply docs/Database_Schema.md]: OKay

File: docs/Database_Schema.md (hunk 2 above)
Comment (MichaelKim20) [reply docs/Database_Schema.md]: How about adding in_index just in case? I think we can print it out as JSON later.

Comment (TrustHenry) [reply docs/Database_Schema.md]: That's a good.

**Comment by TrustHenry** [comment]
@MichaelKim20  Thanks for review. I apply the suggestions.

File: docs/Database_Schema.md [hunk 4]
```diff
@@ -224,3 +224,78 @@ CREATE TABLE IF NOT EXISTS information (
...
+| signature         | BLOB      |    | Y        |          | The signature of this transaction input|
+### _Create Script_
+
+```sql
+CREATE TABLE IF NOT EXISTS "txInputPool" (
+    "tx_hash"               BLOB    NOT NULL,
+    "input_index"           INTEGER NOT NULL,
+    "utxo"                  BLOB    NOT NULL,
+    "signature"             BLOB    NOT NULL,
+    PRIMARY KEY("tx_hash","utxo")
```
Comment (MichaelKim20) [inline docs/Database_Schema.md]:
```suggestion
    PRIMARY KEY("tx_hash","input_index")
```
Isn't this what you want?

Comment (TrustHenry) [reply docs/Database_Schema.md]: Nice catch! Thank you.

**Review by MichaelKim20 — APPROVED** [review]

**Pull request merged**